core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -2016,22 +2015,25 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
-              policyEmails <- DBIO.from(samDAO.listPoliciesForResource(SamResourceTypeNames.workspace, workspaceId, userInfo).map(_.flatMap(policy =>
-                if(policy.policyName == SamWorkspacePolicyNames.projectOwner && workspaceRequest.authorizationDomain.getOrElse(Set.empty).isEmpty) {
-                  // when there isn't an auth domain, we will use the billing project admin policy email directly on workspace
-                  // resources instead of synching an extra group. This helps to keep the number of google groups a user is in below
-                  // the limit of 2000
-                  Option(WorkspaceAccessLevels.ProjectOwner -> projectOwnerGroupEmail)
-                } else {
-                  WorkspaceAccessLevels.withPolicyName(policy.policyName.value).map(_ -> policy.email)
-                }).toMap))
-              _ <- DBIO.from(gcsDAO.setupWorkspace(userInfo, project.get, policyEmails, bucketName, getLabels(workspaceRequest.authorizationDomain.getOrElse(Set.empty).toList)))
-              response <- op(SlickWorkspaceContext(savedWorkspace))
-            } yield response
+          DBIO.from(samDAO.getPolicySyncStatus(SamResourceTypeNames.billingProject, workspaceRequest.namespace, SamBillingProjectPolicyNames.owner, userInfo).map(_.email)).flatMap { projectOwnerPolicyEmail =>
+            saveNewWorkspace(workspaceId, workspaceRequest, bucketName, projectOwnerPolicyEmail, dataAccess).flatMap { savedWorkspace =>
+              for {
+                //there's potential for another perf improvement here for workspaces with auth domains. if a workspace is in an auth domain, we'll already have
+                //the projectOwnerEmail, so we don't need to get it from sam. in a pinch, we could also store the project owner email in the rawls DB since it
+                //will never change, which would eliminate the call to sam entirely
+                policyEmails <- DBIO.from(samDAO.listPoliciesForResource(SamResourceTypeNames.workspace, workspaceId, userInfo).map(_.flatMap(policy =>
+                  if (policy.policyName == SamWorkspacePolicyNames.projectOwner && workspaceRequest.authorizationDomain.getOrElse(Set.empty).isEmpty) {
+                    // when there isn't an auth domain, we will use the billing project admin policy email directly on workspace
+                    // resources instead of synching an extra group. This helps to keep the number of google groups a user is in below
+                    // the limit of 2000
+                    Option(WorkspaceAccessLevels.ProjectOwner -> projectOwnerPolicyEmail)
+                  } else {
+                    WorkspaceAccessLevels.withPolicyName(policy.policyName.value).map(_ -> policy.email)
+                  }).toMap))
+                _ <- DBIO.from(gcsDAO.setupWorkspace(userInfo, RawlsBillingProjectName(workspaceRequest.namespace), policyEmails, bucketName, getLabels(workspaceRequest.authorizationDomain.getOrElse(Set.empty).toList)))
```

Comment:
the smaller change is on this line: we don't need to go to the rawls DB to get the billing project name, we already have it inside of the workspaceRequest

this will likely only have a minor impact on performance (<50ms) but i noticed it while i was digging around and it was a simple thing to change so i did it.